databroker/core.py

```diff
@@ -607,6 +608,31 @@ def stream_gen(entry):
     yield ('stop', stop)
 
 
+class ArgsDict(dict):
+    """
+    This class is made so that we can define our own tokenize method, which
+    greatly speeds up the time to load a BlueskyRun.
+    """
+    ...
+
+# Register a tokenize method for ArgsDict
+@normalize_token.register(ArgsDict)
+def tokenize_dict(args):
+    return (descriptor['uid'] for descriptor in args['metadata']['descriptors'])
```

Comment:
This token must uniquely define the `ArgsDict` for the entire scheduler. Any key collisions between different `ArgsDict` instances are likely to manifest as very confusing bugs. I think I can imagine a scenario where there are two `ArgsDict` instances with the same descriptors but, say, different `self.fillers` or something.

From the discussion on Slack, it looks like the costly step is stringify-ing the descriptors. What sort of performance do you see if you make `args` a plain `dict` but change

```diff
-                metadata={'descriptors': descriptors,
-                          'resources': self._resources})
+                metadata=MetaDict({'descriptors': descriptors,
+                          'resources': self._resources}))
```

If that gives us _almost_ the same speedup, it would be significantly safer and a lighter touch.

Reply:
We could also wrap the start document in a similar subclass.   At this point we have seen the slow down on big descriptors, but it is only a matter of time until we see the slow down on the (up to 16MB) of random user supplied input in the start document :smiling_imp: 

Similarly, we could wrap just the descriptors to source a token that is `(start_uid, desc_uid)`.